### DISCUSSION TITLE
What if Emacs followed the Design Philosophy of Meow With Regards to Defaults?

### DISCUSSION THREAD

**Discussion by jgarte** [meow-edit/meow#483] [Ideas]
> Meow has no default command layout.

What if Emacs itself followed this philosophy as well? Just an idea.

> You can write your own setup function from scratch or pick one from below as your starting point.

Imagine if Emacs started out as a blank slate with no keybindings preconfigured and with way less opinion with regards to its APIs. From that starting point, then, we continue with the Emacs thesis, [as stated](https://www.gnu.org/software/guile/manual/html_node/The-Emacs-Thesis.html).

I'm referring to having an even more radically [bottom up](http://www.paulgraham.com/progbot.html) design than what was actually done. 

One concrete way to have had this would have been for Emacs to have had no keybindings configured at all and to only provide all the built-in Emacs Lisp modules without any predefined keybindings as initially chosen by the original authors.

C'est la vie and RIIR or zig or [hare](https://harelang.org/) in this or another lifetime...

**Comment by DogLooksGood** [comment]
That would be a suckless style editor, I'm really interested in it. But obviously, only minimal editors can be designed in this way.

**Comment by eshrh** [comment]
meow doesn't have that many core commands, which is why it works. Emacs has got thousands, and the default keybinds are honestly kind of a discovery mechanism

**Comment by underside** [comment]
Obviously it's a bad idea, since it's suitable only for expirience users. In Emacs world a lot of people use Emacs with default settings and it's absolutely OK for them. Also, keep in mind that Emacs popular  among not-IT guys, such as writers e.t.c. and even evil-mode is behind of their needs and it's absolutely normal. 
Emacs should leave as it is, at least for default configuration.

But, I think it could be some easier way to re-define  all standart config for power users.
